rust/arrow/src/buffer.rs

```diff
@@ -56,6 +57,23 @@ pub struct Buffer {
 }
 
 impl Buffer {
+    /// Initializes a [Buffer] from a slice of items.
+    pub fn from_slice_ref<U: ArrowNativeType, T: AsRef<[U]>>(items: &T) -> Self {
```

Comment:
This was necessary because it is not possible to have this generic on `impl`, see https://users.rust-lang.org/t/how-to-implement-from-asref-u-for-generic-u/53533/5